crates/ruff_python_formatter/src/expression/string.rs

```diff
@@ -1043,6 +1043,80 @@ impl<'ast, 'buf, 'fmt> DocstringLinePrinter<'ast, 'buf, 'fmt> {
 
         Ok(())
     }
+
+
+/// Represents a single line in a docstring.
+///
+/// This type is used to both represent the original lines in a docstring
+/// (the line will be borrowed) and also the newly formatted lines from code
+/// snippets (the line will be owned).
+#[derive(Clone, Debug)]
+struct DocstringLine<'src> {
+    /// The actual text of the line, not including the line terminator.
+    line: Cow<'src, str>,
```

Comment:
It took me a while to understand when the line is owned or borrowed although it is well documented. It may help to repeat the struct documentation in a short form:

The actual text (Borrowed: original line in the source, Owned: formatted line) of the line

Reply:
Love it. Good catch.